metrics-operator/controllers/analysis/provider_selector.go

```diff
@@ -85,20 +83,18 @@ func (ps ProvidersPool) DispatchToProviders(ctx context.Context, id int) {
 			if err != nil {
 				ps.log.Error(err, "Failed to get Provider")
 				ps.results <- metricsapi.ProviderResult{Objective: j.AnalysisValueTemplateRef, ErrMsg: err.Error()}
-				ps.cancel()
-				return
+				continue
 			}
 
 			templatedQuery, err := generateQuery(templ.Spec.Query, ps.Analysis.Spec.Args)
 			if err != nil {
 				ps.log.Error(err, "Failed to substitute args in templ")
 				ps.results <- metricsapi.ProviderResult{Objective: j.AnalysisValueTemplateRef, ErrMsg: err.Error()}
-				ps.cancel()
-				return
+				continue
 			}
 			//send job to provider solver
 			ps.providers[providerRef.Spec.Type] <- metricstypes.ProviderRequest{
-				Objective: &j,
+				Objective: j,
```

Comment:
This needed to be changed to a value instead of pointer, as otherwise `j` would point to the wrong object in case there are multiple objectives